--- origsrc/FreeImage/Source/Metadata/Exif.cpp	2013-12-31 19:28:28.158825000 -0600
+++ src/FreeImage/Source/Metadata/Exif.cpp	2013-12-31 19:39:21.924427200 -0600
@@ -495,7 +495,7 @@ jpeg_read_exif_dir(FIBITMAP *dib, const
 		}
 
 		// remember that we've visited this directory so that we don't visit it again later
-		DWORD visited = (((DWORD)((long)ifdp) & 0xFFFF) << 16) | (DWORD)de;
+		DWORD visited = (((DWORD)((long long)ifdp) & 0xFFFF) << 16) | (DWORD)de;
 		if(visitedIFD.find(visited) != visitedIFD.end()) {
 			continue;
 		} else {
